src/app/loadout-builder/generated-sets/GeneratedSetButtons.tsx

```diff
@@ -46,9 +46,9 @@ export default function GeneratedSetButtons({
 
   const statsWithPlus5: number[] = [];
 
-  for (const [stat, value] of Object.entries(set.stats)) {
+  for (const [statHash, value] of Object.entries(set.stats)) {
     if (value % 10 > 4) {
-      statsWithPlus5.push(statHashes[stat]);
+      statsWithPlus5.push(parseInt(statHash, 10));
```

Comment:
The fact that these hashes are strings in some places and numbers in others is odd. I think that is on the API thought from my very brief investigation.

Reply:
No, they're always numbers. But keys of objects in JavaScript are always strings. We fake it most of the time based on the property that if you use a string in number context it'll turn into a number, but many things are correctly typed in TS as considering object properties as strings. This means sometimes you need to make the conversion explicit. 

Another approach would be to switch to `Map<number, whatever>` which is more typesafe and maaaybe more performant (but also maybe slower) but it's definitely harder to use. Worth an experiment to see if it improves performance at some point.